kyaml/yaml/merge3/multiplemergekeys_test.go

```diff
@@ -0,0 +1,343 @@
+package merge3_test
+
+var multipleMergeKeysTestCases = []testCase{
+	//
+	// Test Case
+	//
+	{
+		description: `Update protocol for a port`,
+		origin: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: test-deployment
+spec:
+  template:
+    spec:
+      containers:
+      - image: test-image
+        name: test-deployment
+        ports:
+        - containerPort: 8080
+          protocol: UDP
+`,
+		update: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: test-deployment
+spec:
+  template:
+    spec:
+      containers:
+      - image: test-image
+        name: test-deployment
+        ports:
+        - containerPort: 8080
+          protocol: TCP
+`,
+		local: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: test-deployment
+spec:
+  template:
+    spec:
+      containers:
+      - image: test-image
+        name: test-deployment
+        ports:
+        - containerPort: 8080
+          protocol: UDP
+`,
+		expected: `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: test-deployment
+spec:
+  template:
+    spec:
+      containers:
+      - image: test-image
+        name: test-deployment
+        ports:
+        - containerPort: 8080
+          protocol: TCP
+`},
+
+	{
+		description: `Retain local protocol`,
```

Comment:
This one's behavior is going to change - the expected output is going to have both containerPort: 8080, protocol: HTTP and containerPort: 8080, protocol: TCP. Is that going to cause problems in kpt?

Reply:
As far as I can tell, the other ones should remain unchanged when #3111 is done

Reply:
It boils down to what user expects. The fundamental question is why is a tuple {containerPort, protocol} being considered as a merge key. Why can't it just be containerPort here and update the protocol, as opposed to appending new one.

Reply:
the merge key for the ports list is {containerPort, protocol} per apimachinery schema.  #3111 

as an aside, that means one cannot do port name aliasing at this time